File: content/blocks/typescript/caesar-cipher.ts
function caesar(text: string, shift: number): string {
  const offset = ((shift % 26) + 26) % 26;
  return text.replace(/[a-z]/gi, (letter) => {
    const base = letter <= "Z" ? 65 : 97;
    const code = letter.charCodeAt(0) - base;
    return String.fromCharCode(((code + offset) % 26) + base);
  });
}

File: content/blocks/go/caesar-cipher.go
func caesar(text string, shift int) string {
	shift = (shift%26 + 26) % 26
	return strings.Map(func(r rune) rune {
		switch {
		case r >= 'a' && r <= 'z':
			return 'a' + (r-'a'+rune(shift))%26
		case r >= 'A' && r <= 'Z':
			return 'A' + (r-'A'+rune(shift))%26
		default:
			return r
		}
	}, text)
}
